services/subscriber/service.go

```diff
@@ -219,19 +182,82 @@ func (s *Service) Points() chan<- *coordinator.WritePointsRequest {
 }
 
 // read points off chan and write them
-func (s *Service) writePoints() {
+func (s *Service) run() {
 	defer s.wg.Done()
-	for p := range s.points {
-		for se, sub := range s.subs {
-			if p.Database == se.db && p.RetentionPolicy == se.rp {
-				err := sub.WritePoints(p)
+	subs := make(map[subEntry]PointsWriter)
+	// Perform initial update
+	s.updateSubs(subs)
+	for {
+		select {
+		case <-s.update:
+			err := s.updateSubs(subs)
+			if err != nil {
+				s.Logger.Println("failed to update subscriptions:", err)
+			}
+		case p, ok := <-s.points:
+			if !ok {
+				return
+			}
+			for se, sub := range subs {
+				if p.Database == se.db && p.RetentionPolicy == se.rp {
+					err := sub.WritePoints(p)
```

Comment:
>  Also, it looks like each subscription will write out batches separately, magnifying the resource consumption per-write with each additional subscription.

Not sure what you are referring to here. Could you clarify?

Would a new go routine per read off `s.points` be too much? Maybe each PointsWriter should have its own channel and go routine?